lib/get_parser.py

```diff
@@ -44,9 +44,37 @@ def __call__(
         setattr(namespace, self.dest, values)
 
 
-def get_parser() -> argparse.ArgumentParser:
-    """build the argument parser and return it"""
+def get_base_parser() -> argparse.ArgumentParser:
```

Comment:
Yeah, I really don't like this as a long-term solution, but I understand why you did it. IMO, the proper way to do it is to have this base_parser, and have each executable only build-in the arguments it needs on top of the base parser, in the executable files themselves.  I'm fine with making that a separate ticket if you agree.

Reply:
OK. I'll push the other changes so you can review, then rebase again.

Reply:
Everything looks good - go ahead and rebase please.